### ISSUE TITLE
[SC-IDE] order of tabs in Editor gets reversed with every launch of SC

### ISSUE THREAD

**Issue by miczac** [supercollider/supercollider#3941]
* SuperCollider version:  master [ 0f779a492]
* Operating system and version:  macos 10.12.6 
* Qt 5.11.1 via homebrew

Order of tabs in in the current session  representing multiple open Documents (files) gets out of order after closing SC and starting again.
It appears the order gets _reversed_ every time SC-IDE is closed and re-launched.
On the contrary the order in the Documents-Docklet is reverse to the order of the Editor's tabs.
And all, but the first file/tab!

In my particular session I have currently 35 files open.

Steps to reproduce
-----------------------------
* move the current profile ```/Users/mz/Library/Application\ Support/SuperCollider``` away
* start SC-IDE
* open 10 arbitrary files
* observe Documents-Docklet and Editor's tabs
* close SC, launch again, check tabs and Doc-Decklet

After posting this observation on sc-users two other observations in that context were reported.
At least one was confirmed to happen on Linux Mint.

**Comment by miczac** [comment]
Additional observation: 
It's enough to open just three files to reproduce this and just reload the current session without having to re-launch SC-IDE:

![screen shot 2018-08-04 at 18 54 21](https://user-images.githubusercontent.com/1922103/43678722-1347f870-9819-11e8-8aff-41419538f743.png)

![screen shot 2018-08-04 at 18 54 58](https://user-images.githubusercontent.com/1922103/43678726-16203df0-9819-11e8-90c2-eadc69db6329.png)

**Comment by buscon** [comment]
I can confirm exactly the same behavior on:

    SuperCollider version: master [ 0f779a4]
    Operating system and version: Linux Mint 18.3 
    Qt 5.5.1

**Comment by brianlheim** [comment]
A result of https://github.com/supercollider/supercollider/pull/3053 - new tabs are opened to the right of the "current tab". Since the current tab is not updated until after all tabs from a session are added, the second, third, etc. tabs are sequentially added to the right of the first and results in this ordering.

**Comment by miczac** [comment]
applied patch, looks good, from user-POV. Thanks!!

**Comment by miczac** [comment]
ok, found something:
creating a new doc via cmd-n somewhere in the middle of the tab-row inserts a "untitled"-tab and selects it as expected.
Not so when the current tab is the last one: 
"Untitled" is appended right-most to the row of tabs but the previous tab, from which cmd-n was sent, stays selected. So I'd have to click on the new "Untitled" to work with it.
I tested it with another, rather large session and there it won't happen.
I can't see a pattern by now why's that.

**Comment by brianlheim** [comment]
@miczac - are you talking about PR #3942? I cannot reproduce what you are describing. When I open a new tab while the rightmost tab is selected, focus moves to the new tab.

**Comment by brianlheim** [comment]
Also, probably best to comment on the PR rather than the issue with PR-specific feedback.

**Comment by miczac** [comment]
yeah, sorry, mixed that up - need to get re-used to github again. :unamused:

**Comment by miczac** [comment]
any objections to merge this? This appears to work and _really_ helps!
